src/types/quote/header.rs

```
@@ -64,10 +63,6 @@ impl TryFrom<[u8; std::mem::size_of::<QuoteHeader>()]> for QuoteHeader {
         let quote_header =
             <Self as zerocopy::FromBytes>::read_from(&value).expect("failed to read quote header");
 
-        if quote_header.version.get() != QUOTE_V3 && quote_header.version.get() != QUOTE_V4 {
```

Do you think would be a good idea to keep this check ? And V5 ? That way we may fail early, as in most cases we parse the header first

---

the version ultimately dictates the structure of the quote body and the quote signature data. 

this check is implicitly performed when it attempts to parse the quote body, as seen [here](https://github.com/automata-network/dcap-rs/blob/b199db2311194c29ece4c0bb64fa14d1ef664afe/src/types/quote/mod.rs#L42-L98).

along with the quote signature parser shown [here](https://github.com/automata-network/dcap-rs/blob/b199db2311194c29ece4c0bb64fa14d1ef664afe/src/types/quote/signature.rs#L49-L56)

adding this explicit check in the header imo means we also need to monitor the changes here as we introduce more versions that we plan on supporting, which i think is redundant.

@smilenow any thoughts on this?

---

I agree with Udit that we should keep the check here, because it will keep the function integrity unless you mark the QuoteHeader type as internal(crate itself) use only, then you can postpone the check in the later function.

---

Resolved https://github.com/automata-network/dcap-rs/pull/21/commits/3768a47813b69bd3489037c1523e7c8ea19839f9